crates/sui/src/upgrade_compatibility.rs

```diff
@@ -1683,29 +1644,57 @@ fn struct_ability_mismatch_diag(
     Ok(diags)
 }
 
-/// Return a diagnostic for an ability mismatch. returns (full version, name, type)
-fn field_to_string(field: &Field) -> (String, String, String) {
-    let mut field_full = format!("'{}: {}'", field.name, field.type_);
-    let mut field_name = format!("'{}'", field.name);
-    let field_type = format!("'{}'", field.type_);
+struct FieldStrings {
+    full: String,
+    name: String,
+    type_: String,
+    pos: String,
+}
+
+/// Format fields into a string to be displayed returns FieldStrings{full version, name, type}
+/// handles positional fields and type params
+fn format_field(field: &Field, type_params: Vec<SourceName>) -> FieldStrings {
+    let field_name = field.name.to_string();
+    let field_type = format_param(&field.type_, type_params.clone(), &mut Vec::new()).unwrap();
+    let mut field_full = format!("{}: {}", field_name, field_type);
+    let mut pos = "".to_string();
 
     if let Some(pos_num) = Regex::new(r"^pos(\d)+$")
```

Comment:
Is this regex correct? What happens if you have more than 9 positional fields? (Let's add a test for this)

```suggestion
    if let Some(pos_num) = Regex::new(r"^pos(\d+)$")
```

Reply:
both work since its a single character set `\d` whether you add the + before or after the capture group `()` wouldn't matter. I'll add that test though!

Reply:
ah yeah you're right I didn't consider the capture group behavior.